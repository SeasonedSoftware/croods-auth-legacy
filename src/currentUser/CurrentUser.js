import React from 'react'
import { Info } from 'croods'

import render from './render'

export default props => (
  <Info
    name={`${props.reducer}.currentUser`}
    id="currentUser"
    path="/auth/validate_token"
    renderError={false}
    renderLoading={false}
    parseResponse={({ data }) => ({ info: data })}
    render={render(props)}
  />
)
