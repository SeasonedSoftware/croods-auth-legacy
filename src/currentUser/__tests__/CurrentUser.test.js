import React from 'react'
import renderer from 'react-test-renderer'

import CurrentUser from '../CurrentUser'

jest.mock('croods', () => ({
  Info: props => <div {...props}>{props.render(props)}</div>,
}))

jest.mock('../render', () => props => info => (
  <div {...props} {...info}>
    renderInfo
  </div>
))

const props = { foo: 'bar', reducer: 'auth' }
const tree = renderer.create(<CurrentUser {...props} />)

it('renders correctly', () => {
  expect(tree.toJSON()).toMatchSnapshot()
})

it('parses response correctly', () => {
  const data = { important: true }
  const response = {
    foo: 'bar',
    data,
  }

  const parser = tree.root.findByProps({ name: 'auth.currentUser' }).props
    .parseResponse
  expect(parser(response)).toMatchObject({ info: data })
})
