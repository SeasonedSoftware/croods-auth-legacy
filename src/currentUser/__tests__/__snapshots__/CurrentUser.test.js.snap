// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="currentUser"
  name="auth.currentUser"
  parseResponse={[Function]}
  path="/auth/validate_token"
  render={[Function]}
  renderError={false}
  renderLoading={false}
>
  <div
    foo="bar"
    id="currentUser"
    name="auth.currentUser"
    parseResponse={[Function]}
    path="/auth/validate_token"
    reducer="auth"
    render={[Function]}
    renderError={false}
    renderLoading={false}
  >
    renderInfo
  </div>
</div>
`;
